subsys/spm/spm.c

```diff
@@ -160,6 +160,41 @@ extern int irq_target_state_is_secure(unsigned int irq);
 	((SPU_PERIPHID_PERM_LOCK_Locked << SPU_PERIPHID_PERM_LOCK_Pos) &       \
 	 SPU_PERIPHID_PERM_LOCK_Msk)
 
+#define FLASH_NSC_REGION(reg_number)                                           \
+	((reg_number << SPU_FLASHNSC_REGION_REGION_Pos) &                      \
+	SPU_FLASHNSC_REGION_REGION_Msk)
+
+#define FLASH_NSC_REGION_LOCK                                                  \
+	((SPU_FLASHNSC_REGION_LOCK_Locked << SPU_FLASHNSC_REGION_LOCK_Pos) &   \
+	SPU_FLASHNSC_REGION_LOCK_Msk)
+
+#define FLASH_NSC_SIZE(reg_size)                                               \
+	((reg_size << SPU_FLASHNSC_SIZE_SIZE_Pos) &                            \
+	SPU_FLASHNSC_SIZE_SIZE_Msk)
+
+#define FLASH_NSC_SIZE_LOCK                                                    \
+	((SPU_FLASHNSC_SIZE_LOCK_Locked << SPU_FLASHNSC_SIZE_LOCK_Pos) &       \
+	SPU_FLASHNSC_SIZE_LOCK_Msk)
+
+#if defined(CONFIG_ARM_FIRMWARE_HAS_SECURE_ENTRY_FUNCS)
+
+static void spm_config_ncs_flash(void)
+{
+	PRINT("Flash NSC region configured\n");
+
+	/* Configure a single region in Secure Flash as Non-Secure Callable
+	 * (NSC) area.
+	 *
+	 * Area to configure: the upper 1kB of the Secure Flash area.
+	 *
+	 * Note: any Secure Entry functions, exposing secure services to the
+	 * Non-Secure firmware, shall be located inside this NSC area.
+	 */
+	NRF_SPU->FLASHNSC[0].REGION = FLASH_NSC_REGION(5);
```

Comment:
The last secure region is region 7 (`0x38000` to `0x40000` = 256k, line 206 in the right-side diff). Because you mention the area to configure is the "upper 1kB of Secure Flash Area", I expected that to be NCS region 7 and `CONFIG_ARM_NSC_REGION_BASE_ADDRESS` to be `0x3FC00` instead of `0x2FC00`. Could you please double check these values?

Reply:
Ah, I the configuration is "correct", the comment needs update, thanks.
We really need to do this properly - see my introductory comment in the PR description.
For now, I might just add an explanation for this hard-coded placement...